server/routes/api/v1/users.js

```diff
@@ -71,19 +44,58 @@ router.get('/:id',
     let id = req.params.id;
     Users.findById(id, (err, user) => {
       if (err) {
-        res.status(500).json({
-          error: (process.env.NODE_ENV === 'production') ? {} : err,
-          message: err.message
-        });
+        res.status(500).json(errorToJson(err));
       } else if (!user) {
-        res.status(404).json({
-          error: (process.env.NODE_ENV === 'production') ? {} : err,
-          message: err.message
-        });
+        res.status(404).json(errorToJson(new Error('User not found')));
       } else {
         res.json(anonymiseUser(user));
       }
     });
   });
 
+router.put('/:id',
+  require('connect-ensure-login').ensureLoggedIn(),
+  isOwner,
+  (req, res) => {
+    let id = req.params.id;
+    let document = req.body;
+    delete document.hash;
+    delete document.salt;
+    delete document.linkedin;
+
+    Users.findOneAndUpdate({_id: id}, document, (err, user) => {
```

Comment:
The API is returning the old version [because mongoose is](https://stackoverflow.com/questions/32811510/mongoose-findoneandupdate-doesnt-return-updated-document). By adding this option to the query the new object is returned which seems like the intended behaviour.
```suggestion
    Users.findOneAndUpdate({_id: id}, document, { new: true }, (err, user) => {
```

Maybe we should implement this for all update routes.